src/content/language/collections.md

```diff
@@ -242,78 +252,347 @@ final constantMap = const {2: 'helium', 10: 'neon', 18: 'argon'};
+`extraOptions` parameter is nullable and the
+spread operator used on `extraOptions` is not null-aware.
+
+<?code-excerpt "misc/test/language_tour/collections/null_spread_operator_in_collection_b.dart (code_sample)"?>
+```dart tag=fails-sa
+List<String> buildCommandLine(
+  String executable,
+  List<String> options, [
+  List<String>? extraOptions,
+]) {
+  return [
+    executable,
+    ...options,
+    ...extraOptions, // <-- OK now.
+  ];
+}
+
+// Usage:
+//   buildCommandLine('dart', ['run', 'my_script.dart'], null);
+// Result:
+//   [dart, run, my_script.dart]
+```
+
+If you want to spread a nullable collection, use a
+null-aware spread element. The following example is valid
+because the null-aware spread operator is used on `extraOptions`.
 
-<?code-excerpt "misc/test/language_tour/built_in_types_test.dart (list-for)"?>
+<?code-excerpt "misc/test/language_tour/collections/null_spread_operator_in_collection_a.dart (code_sample)"?>
 ```dart
-var listOfInts = [1, 2, 3];
-var listOfStrings = ['#0', for (var i in listOfInts) '#$i'];
-assert(listOfStrings[1] == '#1');
+List<String> buildCommandLine(
+  String executable,
+  List<String> options, [
+  List<String>? extraOptions,
+]) {
+  return [
+    executable,
+    ...options,
+    ...?extraOptions, // <-- OK now.
+  ];
+}
+
+// Usage:
+//   buildCommandLine('dart', ['run', 'my_script.dart'], null);
+// Result:
+//   [dart, run, my_script.dart]
 ```
 
-For more details and examples of using collection `if` and `for`, see the
-[control flow collections proposal.][collections proposal]
+To learn more about the null-aware spread operator, see
+[Spread operator][].
 
-[collections]: /libraries/dart-core#collections
-[type inference]: /language/type-system#type-inference
-[`List`]: {{site.dart-api}}/dart-core/List-class.html
-[`Map`]: {{site.dart-api}}/dart-core/Map-class.html
-[Using constructors]: /language/classes#using-constructors
-[collections proposal]: {{site.repo.dart.lang}}/blob/main/accepted/2.3/control-flow-collections/feature-specification.md
-[spread proposal]: {{site.repo.dart.lang}}/blob/main/accepted/2.3/spread-collections/feature-specification.md
-[generics]: /language/generics
-[`Set`]: {{site.dart-api}}/dart-core/Set-class.html
-[if-case]: /language/branches#if-case
+[Spread operator]: /language/operators/#spread-operators
+
+### If elements {: #if-element }
+
+You can use an `if` element inside of a collection
+to conditionally include elements in the collection. The
+`if` element has a few syntax variations:
+
+```dart
+// If the bool expression is true, include the result.
+if (<bool_expression>) <result>
+```
+
+```dart
+// If the expression matches the pattern, include the result.
+if (<expression> case <pattern>) <result>
+```
+
+```dart
+// If the operation such as '(<bool_expression>)'
+// or '(<expression> case <pattern>)' resolves as true,
+// include the first result, otherwise include the second
+// result.
```

Comment:
Consider adjusting the break here to better collocate the "otherwise" clause.

```suggestion
// include the first result, otherwise,
// include the second result.
```